roles/vdm/library/siteconfig_info.py

```diff
@@ -62,7 +62,7 @@ def traverse(self, folder):
           yamlblock = yaml.safe_load(file)
           # handle components
           if "kind" in yamlblock and yamlblock['kind'] == "Component":
-            self.add_overlays(Overlay.COMPONENT, kustomizefilefullpath)
+            self.add_overlays(Overlay.COMPONENT, folder)
```

Comment:
The code now forces all component level items to be in their own folder with this code update correct? Not saying this is a bad thing, but we have to make sure any older code that might have included a component that lives in say for example: site-config/foo/mycomponent.yaml file is also still processed correctly if this is a viable example.

This would be a major change as we're now altering how DAC works within the structure of the site-config directory.

Reply:
It does require that component level items be in their own folder which I do not believe is a bad thing. Checking the examples of components that Viya provides, there are 18 of those under (security, sas-cas-server and sas-singlestore, in my order) all follow the pattern of providing a component within a folder and that folder contains one kustomization.yaml file.
Based on experience of what a reference should look like for a component or resource in a base kustomization.yaml file, the base kustomization.yaml entries for components that were created before the change should generate a kustomize error (generating a manifest) if someone had tried to include a different component by copying it into the site-config folder structure. I'll run a test to verify that conclusion.

Reply:
If the previous items were setup under their own component directories was this working or do we have hardcoded items in DAC picking these items up or adding the entries into the kustomization.yaml file? Not sure we were actually processing these directories. Let me know what you find

Reply:
yes, we have hardcoded items in DAC placing the correct entries for components in the base kustomization.yaml file, TLS is one example, so those are not using the code in siteconfig_info that was generating the improper entries.

Reply:
I posted some test results in our team chat.